chain/chain-primitives/src/error.rs

```diff
@@ -229,157 +221,111 @@ pub trait LogTransientStorageError {
-            | ErrorKind::InvalidStateRequest(_)
-            | ErrorKind::InvalidRandomnessBeaconOutput
-            | ErrorKind::InvalidBlockMerkleRoot
-            | ErrorKind::NotAValidator
-            | ErrorKind::InvalidChallengeRoot => true,
+        match self {
+            Error::BlockKnown(_)
+            | Error::Orphan
+            | Error::ChunkMissing(_)
+            | Error::ChunksMissing(_)
+            | Error::InvalidChunkHeight
+            | Error::IOErr(_)
+            | Error::Other(_)
+            | Error::ValidatorError(_)
+            | Error::EpochOutOfBounds(_)
+            | Error::ChallengedBlockOnChain
+            | Error::StorageError(_)
+            | Error::GCError(_)
+            | Error::DBNotFoundErr(_) => false,
+            Error::InvalidBlockPastTime(_, _)
+            | Error::InvalidBlockFutureTime(_)
+            | Error::InvalidBlockHeight(_)
+            | Error::InvalidBlockProposer
+            | Error::InvalidChunk
+            | Error::InvalidChunkProofs(_)
+            | Error::InvalidChunkState(_)
+            | Error::InvalidChunkMask
+            | Error::InvalidStateRoot
+            | Error::InvalidTxRoot
+            | Error::InvalidChunkReceiptsRoot
+            | Error::InvalidOutcomesProof
+            | Error::InvalidChunkHeadersRoot
+            | Error::InvalidChunkTxRoot
+            | Error::InvalidReceiptsProof
+            | Error::InvalidStatePayload
+            | Error::InvalidTransactions
+            | Error::InvalidChallenge
+            | Error::MaliciousChallenge
+            | Error::IncorrectNumberOfChunkHeaders
+            | Error::InvalidEpochHash
+            | Error::InvalidNextBPHash
+            | Error::NotEnoughApprovals
+            | Error::InvalidFinalityInfo
+            | Error::InvalidValidatorProposals
+            | Error::InvalidSignature
+            | Error::InvalidApprovals
+            | Error::InvalidGasLimit
+            | Error::InvalidGasPrice
+            | Error::InvalidGasUsed
+            | Error::InvalidBalanceBurnt
+            | Error::InvalidShardId(_)
+            | Error::InvalidStateRequest(_)
+            | Error::InvalidRandomnessBeaconOutput
+            | Error::InvalidBlockMerkleRoot
+            | Error::NotAValidator
+            | Error::InvalidChallengeRoot => true,
         }
     }
 
     pub fn is_error(&self) -> bool {
-        match self.kind() {
-            ErrorKind::IOErr(_) | ErrorKind::Other(_) | ErrorKind::DBNotFoundErr(_) => true,
+        match self {
+            Error::IOErr(_) | Error::Other(_) | Error::DBNotFoundErr(_) => true,
             _ => false,
         }
     }
 }
 
-impl From<ErrorKind> for Error {
-    fn from(kind: ErrorKind) -> Self {
-        Self { inner: anyhow::Error::new(kind) }
-    }
-}
-
 impl From<io::Error> for Error {
     fn from(error: io::Error) -> Self {
-        Self { inner: anyhow::Error::new(ErrorKind::IOErr(error.to_string())) }
+        Error::IOErr(error.to_string())
```

Comment:
Should we just change IOErr to hold std::io::Error rather than String?

Also, do we want to repackage std::io::ErrorKind::Other into our own Other?

Reply:
>Should we just change IOErr to hold std::io::Error rather than String?

We should, but we can't trivially -- Error is Eq, while io::Error is !Eq. Errors shouldn't generally be Eq, but we use that in tests. 

> Also, do we want to repackage std::io::ErrorKind::Other into our own Other?

Probably not -- there's also io::ErrorKind::Unclassified. My gut feeling is that we want to remove Error::Other altogether, but that requireds understanding of how we use the type. In general, this feels to me like a kitchen-sink error, and that's not great.

Reply:
Thinking about this more, why not? 

https://github.com/near/nearcore/pull/6806